### PR TITLE
Merged from wince_platform of sdl_core repository.

### PR DESCRIPTION
**Pull request by VENGEL**
1. Solved the crash problem when connect from mobile to sdl.
2. Solved the problem that failed to register the app after connection done.
3. Increase the buffer size of socket.
4. Solved all the failures of unit test on wince platform.

### CODE REVIEW

File: src/components/protocol_handler/test/protocol_handler_tm_test.cc [hunk 1]
```diff
@@ -931,6 +931,9 @@ TEST_F(ProtocolHandlerImplTest, FloodVerification) {
                   message_id,
                   &some_data[0]);
   }
+#ifdef OS_WIN32
+  Sleep(1000);
```
Comment (BrandonHe) [inline src/components/protocol_handler/test/protocol_handler_tm_test.cc:935]: Here add below add many `Sleep()` for  OS_WIN32 port, it's just for Win32 port? Why add this timer delay? if necessary, please add comments for other developers.

File: src/components/protocol_handler/test/protocol_payload_test.cc [hunk 2]
```diff
@@ -258,8 +258,9 @@ TEST(ProtocolPayloadTest, ExtractProtocolWithJSONWithDataWithWrongPayloadSize) {
   prot_payload_test.json = expect_output_json_string;
   prot_payload_test.header.json_size = prot_payload_test.json.length();
 
-  const size_t data_for_sending_size =
-      PROTOCOL_HEADER_V2_SIZE + prot_payload_test.json.length();
+  const size_t data_for_sending_size = PROTOCOL_HEADER_V2_SIZE +
+                                       prot_payload_test.json.length() + 
+                                       prot_payload_test.data.size();
```
Comment (BrandonHe) [inline src/components/protocol_handler/test/protocol_payload_test.cc:263]: @VENGEL Please confirm here why add `data.size()`. I notice it not happens on SmartDeviceLink Repo

**Comment by BrandonHe** [comment]
@VENGEL @smartdevice475 Do you have passed the test?

File: src/components/protocol_handler/test/protocol_payload_test.cc (hunk 2 above)
Comment (VENGEL) [reply src/components/protocol_handler/test/protocol_payload_test.cc:263]: 这个之前邮件说过，北美最新代码是这样的

File: src/components/protocol_handler/test/protocol_handler_tm_test.cc (hunk 1 above)
Comment (VENGEL) [reply src/components/protocol_handler/test/protocol_handler_tm_test.cc:935]: win32平台速度快，两个线程不同步，导致访问非法内存崩溃，wince平台没有这个问题

**Comment by VENGEL** [comment]
测试通过，多次确认OK

**Comment by BrandonHe** [comment]
@VENGEL 后面发pull request请不要直接从`VENGEL:master`,正确的流程是，基于master分支，建立新的分支，并且新的分支以`hotfix/resolve_bug_xxx`命名， `xxx`为你要解决的issue

**Comment by VENGEL** [comment]
之后这边github的分支管理会出一个执行方案，目前没有，可能下周会跟你讨论

**Pull request merged**